src/scripts/main.js

```diff
@@ -1,3 +1,170 @@
+    return;
+  }
+
+  if (LastClickedTarget !== e.target) {
+    clickCounter = 0;
+  }
+  LastClickedTarget = e.target;
+  clickCounter++;
+
+  const rows = Array.from(tbody.querySelectorAll('tr'));
+
+  if (clickCounter === 1) {
+    rows.sort((a, b) => {
+      const cell1 = a.cells[e.target.cellIndex].textContent;
+      const cell2 = b.cells[e.target.cellIndex].textContent;
+
+      if (e.target.textContent === 'Salary') {
+        const num1 = cell1.replace(/[$,]/g, '');
+        const num2 = cell2.replace(/[$,]/g, '');
+
+        return num1 - num2;
+      }
+
+      return cell1.localeCompare(cell2);
+    });
+  }
+
+  if (clickCounter === 2) {
+    clickCounter = 0;
+
+    rows.sort((a, b) => {
+      const cell1 = a.cells[e.target.cellIndex].textContent;
+      const cell2 = b.cells[e.target.cellIndex].textContent;
+
+      if (e.target.textContent === 'Salary') {
+        const num1 = cell1.replace(/[$,]/g, '');
+        const num2 = cell2.replace(/[$,]/g, '');
+
+        return num2 - num1;
+      }
+
+      return cell2.localeCompare(cell1);
+    });
+  }
+  tbody.append(...rows);
+});
+
+// selecting
+tbody.addEventListener('click', (e) => {
+  if (document.querySelector('.active')) {
+    document.querySelector('.active').classList.remove('active');
+  }
+
+  e.target.closest('tr').classList.add('active');
+});
+
+const form = document.createElement('form');
+
+form.classList.add('new-employee-form');
+
+form.innerHTML = `
+<label>Name: 
+  <input 
+    name="name" 
+    type="text" 
+    data-qa="name" 
+    minlength="4"
+    required
+  >
+</label>
+
+<label>Position: 
+  <input 
+    name="position" 
+    type="text" 
+    data-qa="position" 
+    required
+  >
+</label>
+
+<label>Office: 
+  <select name="office" data-qa="office">
+    <option value="Tokyo">Tokyo</option>
+    <option value="Singapore" selected>Singapore</option>
+    <option value="London">London</option>
+    <option value="New York">New York</option>
+    <option value="Edinburgh">Edinburgh</option>
+    <option value="San Francisco">San Francisco</option>
+  </select>
+</label>
+
+<label>Age: 
+  <input 
+    name="age" 
+    type="number" 
+    data-qa="age" 
+    min="18" 
+    max="90"
+    required
+  >
+</label>
+
+<label>Salary: 
+  <input 
+    name="salary" 
+    type="text" 
+    data-qa="salary" 
+    required
+  >
+</label>
+
+<button type="submit">Save to table</button>
+`;
+
+body.append(form);
+
+// form validation & adding info to the table
+document.querySelector('button').addEventListener('click', (e) => {
+  e.preventDefault();
+
+  let validity = true;
+
+  for (const i of form) {
+    if (!i.checkValidity()) {
+      validity = false;
+    }
+  }
+
+  const div = document.createElement('div');
+
+  div.className = 'notification';
+
+  if (validity) {
+    div.textContent = 'SUCCESS';
+    div.classList.add('success');
+
+    const tr = document.createElement('tr');
+
+    for (let i = 0; i < form.length - 1; i++) {
```

Comment:
Use a more descriptive variable name instead of `i` for the loop, like `inputIndex`.